pubsub/google/cloud/pubsub_v1/futures.py

```diff
@@ -29,18 +29,28 @@ class Future(google.api_core.future.Future):
 
     This object should not be created directly, but is returned by other
     methods in this library.
+
+    Args:
+        event_factory (Optional[Callable[[], Any]]): An event factory, expected
+            to take no arguments and return an event with the same interface as
+            :class:`threading.Event`. This is provided so that callers
+            with different concurrency models (e.g. ``threading`` or
+            ``multiprocessing``) can wait on an event that is compatible
+            with that model. The ``wait()`` and ``set()`` methods will be
+            used on the returned event. The default is
+            :class:`threading.Event`.
     """
 
     # This could be a sentinel object or None, but the sentinel object's ID
     # can change if the process is forked, and None has the possibility of
     # actually being a result.
     _SENTINEL = uuid.uuid4()
 
-    def __init__(self):
+    def __init__(self, event_factory=threading.Event):
```

Comment:
Worth noting that doing it like this (vs. `event_factory=None` and then doing an `if event_factor is None` check) makes it much more difficult to mock `threading.Event`.